src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -241,41 +274,97 @@ public void paintBorder(Component c, Graphics g, int x, int y,
+                xtranslation = roundDown(at.getScaleX() * x + at.getTranslateX());
+                ytranslation = roundDown(at.getScaleY() * y + at.getTranslateY());
+            } else {
+                width = w;
+                height = h;
+                xtranslation = x;
+                ytranslation = y;
+            }
+            g2d.translate(xtranslation, ytranslation);
+
+            // border and corner scaling
+            corner = (int) Math.round(corner * at.getScaleX());
+            // loop constraint for bulk of the border
+            int loopCount = (int) Math.round(5 * at.getScaleX());
+
+            // midpoint at which highlight & shadow lines
+            // are positioned on the border
+            int midPoint = loopCount/2;
+
+            g.setColor(background);
+            // Draw outermost lines
+            g.drawLine( 0, 0, width-1, 0);
+            g.drawLine( 0, 1, 0, height-1);
+            g.drawLine( width-1, 1, width-1, height-1);
+            g.drawLine( 1, height-1, width-1, height-1);
+
+            // Draw the bulk of the border
+            for (int i = 1; i <= loopCount; i++) {
+                g.drawRect(x+i,y+i,width-(i*2)-1, height-(i*2)-1);
+            }
+
+            if (c instanceof JInternalFrame && ((JInternalFrame)c).isResizable()) {
+                // Draw the Long highlight lines
+                g.setColor(highlight);
+                g.drawLine(corner+1, midPoint+stkWidth,
+                        width-corner, midPoint+stkWidth); //top
+                g.drawLine(midPoint+stkWidth, corner+1,
+                        midPoint+stkWidth, height-corner); //left
+                g.drawLine(width-midPoint, corner+1,
+                        width-midPoint, height-corner); //right
+                g.drawLine(corner+1, height-midPoint,
+                        width-corner, height-midPoint); //bottom
+
+                // Draw the Long shadow lines
+                g.setColor(shadow);
+                g.drawLine(corner, midPoint, width-corner-1, midPoint);
+                g.drawLine(midPoint, corner, midPoint, height-corner-1);
+                g.drawLine(width-(midPoint+stkWidth), corner,
+                        width-(midPoint+stkWidth), height-corner-1);
+                g.drawLine(corner, height-(midPoint+stkWidth),
+                        width-corner-1, height-(midPoint+stkWidth));
+            }
 
-          }
+            // Undo the resetTransform setting from before
+            g2d.translate(-xtranslation, -ytranslation);
+            if (resetTransform) {
+                g2d.setColor(oldColor);
+                g2d.setTransform(at);
+                g2d.setStroke(oldStk);
+            }
+        }
 
-          public Insets getBorderInsets(Component c, Insets newInsets) {
-              newInsets.set(5, 5, 5, 5);
-              return newInsets;
-          }
+        public Insets getBorderInsets(Component c, Insets newInsets) {
+          newInsets.set(5, 5, 5, 5);
```

Comment:
The default code inset is 4 so this formatting looks incorrect. Plus, i understand that you were trying to better indent the code but usually if code is not changed functionally we avoid changing the lines to make code annotation more informative.